packages/edit-navigation/src/components/layout/style.scss

```diff
@@ -65,12 +65,18 @@
 }
 
 .edit-navigation-layout__block-toolbar {
+	// Make the fixed toolbar appear in a similar position to the floating toolbar.
+	// Take the spacing for the floating toolbar, then subtract the toolbar height and
+	// the gap between the content area and the toolbar.
+	margin-top: $navigation-editor-spacing-top - $block-toolbar-height - $grid-unit-15;
+
 	.block-editor-block-toolbar {
 		background: $white;
 		border: $border-width solid $gray-900;
 		border-radius: $radius-block-ui;
 		max-width: $navigation-editor-width;
 		margin: auto;
+		overflow-y: hidden;
```

Comment:
I think we allow generic overflow to allow the entire toolbar to be horizontally draggable on a physical mobile device. That seems a good enough reason to add this rule here, contextual to the nav screen whose responsiveness seems a separate issue.